src/watertap_contrib/seto/unit_models/surrogate/med_tvc_surrogate.py

```diff
@@ -0,0 +1,1181 @@
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        units_meta = self.config.property_package.get_metadata().get_derived_units
+
+        """
+        Add system configurations
+        """
+        self.Nef = Param(
+            initialize=12, units=pyunits.dimensionless, doc="Number of effects"
+        )
+
+        self.RR = Var(
+            initialize=0.50,
+            bounds=(0.30, 0.50),
+            units=pyunits.dimensionless,
+            doc="Recovery ratio",
+        )
+
+        self.Q_loss = Param(
+            initialize=0.054, units=pyunits.dimensionless, doc="System thermal loss"
+        )
+
+        """
+        Add block for feed water
+        """
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = False
+
+        self.feed_props = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of feed water",
+            **tmp_dict
+        )
+
+        # Set alias for feed water salinity and convert to mg/L for surrogate model
+        self.Xf = pyunits.convert(
+            self.feed_props[0].conc_mass_phase_comp["Liq", "TDS"],
+            to_units=pyunits.mg / pyunits.L,
+        )
+
+        # Set alias for feed water temperature and convert to degree C for surrogate model
+        self.Tin = self.feed_props[0].temperature - 273.15 * pyunits.K
+
+        # Set alias for feed water enthalpy and convert to kJ/kg
+        h_sw = pyunits.convert(
+            self.feed_props[0].enth_mass_phase["Liq"], to_units=pyunits.kJ / pyunits.kg
+        )
+
+        # Set alias for feed water mass flow rate (kg/s)
+        m_f = sum(
+            self.feed_props[0].flow_mass_phase_comp["Liq", j]
+            for j in self.feed_props.component_list
+        )
+
+        """
+        Add block for distillate
+        """
+        self.distillate_props = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of distillate",
+            **tmp_dict
+        )
+
+        # distillate salinity is 0
+        @self.Constraint(doc="distillate salinity")
+        def distillate_s(b):
+            return b.distillate_props[0].conc_mass_phase_comp["Liq", "TDS"] == 0
+
+        # Connect the plant capacity with the distillate mass flow rate
+        self.Capacity = Var(
+            initialize=2000,
+            bounds=(2000, 100000),
+            units=pyunits.m**3 / pyunits.d,
+            doc="Capacity of the plant (m3/day)",
+        )
```

Comment:
Why make this a variable? You can have this as an expression since it seems all you need is flow rate, which is a property.